google/cloud/spanner/client.h

```diff
@@ -0,0 +1,193 @@
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_CLIENT_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_CLIENT_H_
+
+#include "google/cloud/spanner/key.h"
+#include "google/cloud/spanner/mutation.h"
+#include "google/cloud/spanner/partition.h"
+#include "google/cloud/spanner/result_set.h"
+#include "google/cloud/spanner/result_stats.h"
+#include "google/cloud/spanner/row.h"
+#include "google/cloud/spanner/sql_statement.h"
+#include "google/cloud/spanner/transaction.h"
+#include "google/cloud/spanner/value.h"
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include <chrono>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+class Client {
+ public:
+  // move-only
```

Comment:
I think we decided differently in the design doc, but maybe I misread things @devjgm.

Reply:
I think @coryan is right. In the doc, we decided to make Client a "regular type", which means copyable (and moveable). In general, that should probably be the default unless there's an explicit reason not to do so.